modules/sdk-lib-mpc/src/tss/ecdsa-dkls/dkg.ts

```diff
@@ -25,6 +31,49 @@ export class Dkg {
     }
   }
 
+  private _createDKLsRetrofitKeyShare() {
+    if (this.retrofitData) {
+      if (!this.retrofitData.xShare.y || !this.retrofitData.xShare.chaincode || !this.retrofitData.xShare.x) {
+        throw Error('xShare must have a public key, private share value, and a chaincode.');
+      }
+      if (this.retrofitData.bigSiList.length !== this.n - 1) {
+        throw Error("bigSiList should contain the other parties' Si's");
+      }
+      const bigSList: Array<Array<number>> = [];
+      const xiList: Array<Array<number>> = [];
+      let j = 0;
+      for (let i = 0; i < this.n; i++) {
+        if (i === this.partyIdx) {
+          const secp256k1 = new Secp256k1Curve();
+          bigSList.push(
+            Array.from(bigIntToBufferBE(secp256k1.basePointMult(BigInt('0x' + this.retrofitData.xShare.x))))
+          );
+        } else {
+          bigSList.push(Array.from(Buffer.from(this.retrofitData.bigSiList[j], 'hex')));
+          j++;
+        }
+        xiList.push(Array.from(bigIntToBufferBE(BigInt(i + 1), 32)));
+      }
+      const dklsKeyShare = {
```

Comment:
Is there a type defined for this? That would ensure you are not missing any fields.